scipy/_lib/deprecation.py

```diff
@@ -57,7 +60,15 @@ def _sub_module_deprecation(*, sub_package, module, private_module, all,
 
     warnings.warn(message, category=DeprecationWarning, stacklevel=3)
 
-    return getattr(import_module(f"scipy.{sub_package}.{private_module}"), attribute)
+    for module in private_modules:
+        try:
+            return getattr(import_module(f"scipy.{sub_package}.{module}"), attribute)
+        except AttributeError as e:
+            # still raise an error if the attribute isn't in any of the expected
+            # private modules
+            if module is private_modules[-1]:
+                raise e
```

Comment:
Comparing strings with `is` is a bit fragile. Why not `==`?

Reply:
No important reason other than a rushed airport commit!
```suggestion
            if module == private_modules[-1]:
                raise e
```